pkg/reconciler/messaging/channel/channel.go

```diff
@@ -310,14 +314,60 @@ func (r *Reconciler) reconcileTopic(ctx context.Context, channel *v1beta1.Channe
 		Annotations:        resources.GetTopicAnnotations(clusterName),
 	})
 
-	topic, err = r.RunClientSet.InternalV1beta1().Topics(channel.Namespace).Create(t)
-	if err != nil {
-		logging.FromContext(ctx).Desugar().Error("Failed to create Topic", zap.Error(err))
-		r.Recorder.Eventf(channel, corev1.EventTypeWarning, "TopicCreateFailed", "Failed to created Topic %q: %s", topic.Name, err.Error())
-		return nil, err
+	if apierrs.IsNotFound(err) {
```

Comment:
Isn't this still the error for not getting the Topic?

Reply:
I refactored this a bit to make it more consistent with other reconcilers. In general, I think we need to do a big refactor to this old channel code. Is pretty hard to follow...
If the error is notFound, then we create the Topic... 
Previously we were doing something like: `if err != nil && !apierrors.IsNotFound(err)`...  now I'm checking for the positive case of notFound.
Does it make sense?

Reply:
Yes.